x/chainlet/keeper/msg_server_upgrade_chainlet.go

```diff
@@ -26,16 +23,22 @@ func (k msgServer) UpgradeChainlet(goCtx context.Context, msg *types.MsgUpgradeC
 		return &types.MsgUpgradeChainletResponse{}, err
 	}
 
-	if !slices.Contains(ogChainlet.Maintainers, msg.Creator) && msg.Creator != SagaAddress {
-		return nil, fmt.Errorf("address %s not whitelisted for creating or updating chainlet stacks", msg.Creator)
+	if !slices.Contains(ogChainlet.Maintainers, msg.Creator) {
+		return nil, fmt.Errorf("address %s is not a chainlet maintainer", msg.Creator)
 	}
 	majorUpgrade, err := versions.CheckUpgrade(ogChainlet.ChainletStackVersion, msg.StackVersion)
 	if err != nil {
 		return nil, err
 	}
-	// Only this Saga-controlled address is allowed to perform (manual) major upgrades until they're automated using IBC
-	if majorUpgrade && msg.Creator != SagaAddress {
-		return nil, errors.New("major upgrades not implemented")
+	if majorUpgrade {
+		p := k.GetParams(ctx)
+		upgradeDelta := p.MinimumUpgradeHeightDelta + msg.HeightDelta
+		err = k.sendUpgradePlan(ctx, &ogChainlet, ogChainlet.ChainletStackVersion, msg.StackVersion, upgradeDelta)
```

Comment:
could be less error prone to just pass the `ogChainlet` and then access that field in the function instead of passing an individual field separately

```suggestion
		err = k.sendUpgradePlan(ctx, &ogChainlet, ogChainlet.ChainletStackVersion, upgradeDelta)
```